ratatui/src/widgets.rs

```diff
@@ -268,6 +268,74 @@
 //! (`&MyWidget` or `&mut MyWidget`). This provides reusability and automatic [`WidgetRef`] support.
 //! You can optionally implement the consuming version for backward compatibility.
 //!
+//! # Authoring Custom Widget Libraries
+//!
+//! When creating a library of custom widgets for distribution, there are specific considerations
+//! that will make your library more compatible and accessible to a wider range of users. Following
+//! these guidelines will help ensure your widget library works well in various environments and
+//! can be easily integrated into different types of applications.
+//!
+//! ## Depend on `ratatui_core`
+//!
+//! For widget libraries, depend on [`ratatui-core`] instead of the full `ratatui` crate. This
+//! provides all the essential types and traits needed for widget development while avoiding
+//! unnecessary dependencies on backends and other components that widget libraries don't need.
+//!
+//! ## Make Your Crate `no_std` Compatible
+//!
+//! For maximum compatibility, especially in embedded environments, consider making your widget
+//! library `no_std` compatible. This is often easier than you might expect and broadens the range
+//! of projects that can use your widgets.
+//!
+//! To do this, first add the `![no_std]` attribute at the top of your `lib.rs`.
+//!
+//! When working in a `no_std` environment:
+//!
+//! - Use `core::` instead of `std::` for basic functionality
+//! - Use `alloc::` for heap-allocated types like `String`, `Vec`, and `Box`
+//! - Add `extern crate alloc;` to access allocation types
+//!
+//! Here's a complete example of a `no_std` compatible widget:
+//!
+//! ```ignore
+//! #![no_std]
+//!
+//! extern crate alloc;
+//!
+//! use alloc::string::String;
+//!
+//! use ratatui_core::buffer::Buffer;
+//! use ratatui_core::layout::Rect;
+//! use ratatui_core::text::Line;
+//! use ratatui_core::widgets::Widget;
+//!
+//! struct MyWidget {
+//!     content: String,
+//! }
+//!
+//! impl Widget for &MyWidget {
+//!     fn render(self, area: Rect, buf: &mut Buffer) {
+//!         Line::raw(&self.content).render(area, buf);
+//!     }
+//! }
+//! ```
+//!
+//! ## Benefits of This Approach
```

Comment:
I think you are referring to `no_std` here so this should be a sub-section.

```suggestion
//! ### Benefits of This Approach
```